### DISCUSSION TITLE
How do to send to a SOCKS5 proxy with username and password?

### DISCUSSION THREAD

**Discussion by CryAboutIt** [xjasonlyu/tun2socks#131] [Q&A] [answered]
> Proxy Protocols: HTTP/Socks4/Socks5/Shadowsocks with authentication support for remote connections.

Can't figure out how to send to a SOCKS5 proxy with username, pass. My best guess would be the standard socks5://user:pass@ip:port but that doesn't seem to work.

**Comment by CryAboutIt** [comment]
nevermind. I must have configured it wrong or something

**Comment by chromer030** [comment, marked as answer]
Socks4
Socks4 only support USERID auth.
`-proxy socks4://<userid>@server_host:port`

Socks5
`-proxy socks5://<username>:<password>@server_host:port`

**Reply by CryAboutIt** [reply]
yeah that must have not been what I was doing wrong

**Reply by rogelioglz** [reply]
Ah muy bien que sería?
